assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java

```diff
@@ -3514,6 +3515,48 @@ public AssignmentSubmission getSubmission(String assignmentReference, User perso
 		return submission;
 	}
 
+	public Map<User, AssignmentSubmission> getUserSubmissionMap(Assignment a, List<User> users)
+	{
+		Map<User, AssignmentSubmission> userSubmissionMap = new HashMap<>();
+		if (a != null && !CollectionUtils.isEmpty(users))
+		{
+			if (a.isGroup())
+			{
+				try
+				{
+					Site _site = SiteService.getSite(a.getContext());
+					for (User user : users)
+					{
+						AssignmentSubmission submission = null;
+						Collection<Group> groups = (Collection<Group>) _site.getGroupsWithMember(user.getId());
```

Comment:
I'm looking at getSubmission(String assignmentReference, User person) and it looks like it already handles this loop. I feel like there's unnecessary duplication in this block in the grouped case.